numpy/meson.build

```diff
@@ -223,17 +223,19 @@ null_dep = dependency('', required : false)
 atomic_dep = null_dep
 code_non_lockfree = '''
   #include <stdint.h>
+  #include <stddef.h>
+  #include <stdatomic.h>
```

Comment:
```suggestion
```

First, thank you so much for pointing out that this was buggy. I guess I never looked carefully at the meson logs. 

That said, I think we can delete this line, since at least on my aarch64 Ubuntu setup, the code as you've modified it compiles without including `stdatomic.h`. The point of this test is to make sure we can use this code "out of the box" without any special includes or linking explicitly against libatomic.

```
diff --git a/numpy/meson.build b/numpy/meson.build
index 88c4029ada..353f89398c 100644
--- a/numpy/meson.build
+++ b/numpy/meson.build
@@ -223,6 +223,7 @@ null_dep = dependency('', required : false)
 atomic_dep = null_dep
 code_non_lockfree = '''
   #include <stdint.h>
+  #include <stddef.h>
   int main() {
    struct {
      void *p;
@@ -230,10 +231,10 @@ code_non_lockfree = '''
    } x;
    x.p = NULL;
    x.u8v = 0;
-   uint8_t res = __atomic_load_n(x.u8v, __ATOMIC_SEQ_CST);
-   __atomic_store_n(x.u8v, 1, ATOMIC_SEQ_CST);
-   void *p = __atomic_load_n(x.p, __ATOMIC_SEQ_CST);
-   __atomic_store_n((void **)x.p, NULL, __ATOMIC_SEQ_CST)
+   uint8_t res = __atomic_load_n(&x.u8v, __ATOMIC_SEQ_CST);
+   __atomic_store_n(&x.u8v, 1, __ATOMIC_SEQ_CST);
+   void *p = __atomic_load_n((void **)x.p, __ATOMIC_SEQ_CST);
+   __atomic_store_n((void **)x.p, NULL, __ATOMIC_SEQ_CST);
    return 0;
    }
 '''
goldbaum@ubuntuvm:~/Documents/numpy$ cat build/meson-logs/meson-log.txt | grep "Check atomic builtins without"
Checking if "Check atomic builtins without -latomic" : links: YES
```